include/seqan3/argument_parser/exceptions.hpp

```diff
@@ -133,13 +134,13 @@ class validation_failed : public parser_invalid_argument
  * - Reuse of a short or long identifier (must be unique)
  * - Both identifiers must not be empty (one is ok)
  */
-class parser_design_error : public std::logic_error
+class design_error : public std::logic_error
```

Comment:
in the Changelog you mentioned, that this inherits from argument_parser_error, which I would prefer. I also would not inherit the argument_parser_error from invalid_argument but make it it's own exception.

Reply:
> in the Changelog you mentioned, that this inherits from argument_parser_error, which I would prefer.

Good catch

> I also would not inherit the argument_parser_error from invalid_argument but make it it's own exception.

You mean it should inherit from std::exception instead of std::invalid_argument?

Reply:
Yes, because design_error is not an invalid_argument is it? I would think it is an own error for itself.